### PR TITLE
feat(resident-app): DOMA-12031 handle more filters in allResidentBillingReceipts

### PR DESCRIPTION
**Pull request by YEgorLu**
Added definition for Decimal, CalendarDay types in generateQueryWhereInput(), and changed types of period, toPay filters, as they appended invalid values in gql, which in fact would just return errors always, if they would be handled correctly

Changed handling of filters, now we can use more specific filters like "period_gte", not just "period" and "AND" / "OR" statements.

Initially I wanted to generate filters on any type in generateQueryWhereInput(), as we know which field types of keystone we connected and field types has method, which returns their "gql where input". But currently it is impossible due to conflicts with @open-keystone field types. Some fields have /** \@jsx */ pragma in their views, which does not affect condo, but triggers build errors in submodules for some reason.

As I understood /** \@jsx */ is something old and irrelevant now, so can be removed. That move requires pr in @open-condo/open-keystone, with pragma removal, additional babel configuration and test of admin panel, which I don't want to do right now.

## Summary by CodeRabbit

* **New Features**
  * Enhanced filtering for resident billing receipts, allowing users to filter by period and amount using range (greater than, less than) and set membership options.

* **Bug Fixes**
  * Improved handling of nested filters to ensure excluded fields (such as service consumer) are consistently ignored at all levels.

* **Tests**
  * Expanded test coverage for filtering receipts by period, including logical AND/OR combinations and edge cases.
  * Added tests for utility function that removes specified keys from nested filter objects.

* **Documentation**
  * Updated schema documentation to reflect new filtering options and removed outdated pattern-matching filters.

### CODE REVIEW

File: apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js [hunk 1]
```diff
@@ -68,6 +70,27 @@ const getFile = (receipt, contacts) => {
     }
 }
 
+function removeKeysFromObjectDeep (obj, keysToRemove = []) {
```
Comment (dkoviazin) [inline apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js]: looks like utils function
better to move it from this file and add some tests to it

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/schema.ts [hunk 2]
```diff
@@ -81860,21 +81852,13 @@ export type ResidentBillingReceiptWhereInput = {
...
-  toPay_not_contains?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_contains_i?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_ends_with?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_ends_with_i?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_i?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_starts_with?: InputMaybe<Scalars['String']['input']>;
-  toPay_not_starts_with_i?: InputMaybe<Scalars['String']['input']>;
-  toPay_starts_with?: InputMaybe<Scalars['String']['input']>;
-  toPay_starts_with_i?: InputMaybe<Scalars['String']['input']>;
+  toPay_not_in?: InputMaybe<Array<InputMaybe<Scalars['String']['input']>>>;
```
[the remark this answers is not in the document]
Comment (YEgorLu) [reply apps/condo/schema.ts:81861]: there is no such thing as Decimal scalar. Decimal is keystone field type, which uses String scalar in gql

File: apps/condo/schema.graphql [hunk 3]
```diff
@@ -83891,36 +83891,20 @@
...
-  toPay_starts_with_i: String
-  toPay_not_starts_with_i: String
-  toPay_ends_with_i: String
-  toPay_not_ends_with_i: String
+  toPay_lt: String
+  toPay_gt: String
+  toPay_lte: String
+  toPay_gte: String
+  toPay_in: [String]
+  toPay_not_in: [String]
```
[the remark this answers is not in the document]
Comment (YEgorLu) [reply apps/condo/schema.graphql:83907]: there is no such thing as Decimal scalar. Decimal is keystone field type, which uses String scalar in gql

File: apps/condo/schema.graphql [hunk 4]
```diff
@@ -83891,36 +83891,20 @@ input ResidentBillingReceiptWhereInput {
...
-  period_starts_with_i: String
-  period_not_starts_with_i: String
-  period_ends_with_i: String
-  period_not_ends_with_i: String
+  period_lt: String
+  period_gt: String
+  period_lte: String
+  period_gte: String
+  period_in: [String]
+  period_not_in: [String]
```
[the remark this answers is not in the document]
Comment (YEgorLu) [reply apps/condo/schema.graphql:83899]: there is no such thing as CalendarDay scalar. CalendarDay is keystone field type, which uses String scalar in gql

File: apps/condo/schema.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (YEgorLu) [reply apps/condo/schema.ts:81861]: there is no such thing as Decimal scalar. Decimal is keystone field type, which uses String scalar in gql

File: apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js (hunk 1 above)
Comment (YEgorLu) [reply apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js]: Moved to utils in billing domain, as I think it is quite specific to allResidentBillingReceipts query.
After some thought I figured that it isn't good to place it in common, as it is not right way to go. Better to just write good API for custom schemas, instead of trying to parse it with something strange

File: apps/condo/domains/billing/utils/gqlWhereInput.utils.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        return obj
+    }
+    for (const key of Object.keys(obj)) {
+        if (keysToRemove.includes(key)) {
+            delete obj[key]
+            continue
+        }
+        if (key === 'AND' || key === 'OR') {
+            obj[key].forEach(objItem => removeKeysFromObjectDeep(objItem, keysToRemove))
+        }
```
[the remark this answers is not in the document]
Comment (YEgorLu) [reply apps/condo/domains/billing/utils/gqlWhereInput.utils.js:20]: Graphql input is being validated by gql, we can not have AND / OR without array

**Review by vovaaxeapolla — APPROVED** [review]
